gossip/c_block_callbacks.go

```diff
@@ -262,6 +268,14 @@ func consensusCallbackBeginBlockFn(
 							}
 						}
 					}
+					// Skipped txs require original order from events
+					skippedTxs := make([]uint32, 0, len(orderedTxs))
+					for _, txNumber := range skippedByEvm {
+						txHash := orderedTxs[txNumber].Hash()
+						skippedTxs = append(skippedTxs, txPositions[txHash].EventOffset)
```

Comment:
The `EventOffset` is, I think, the position of the transaction in the list of transactions of the first event where it is showing up. For the resulting skip list should reference the position of skipped transactions in the list of transactions one would obtain by merging the internal transactions, followed by explicit transactions, followed by the transactions from the events. An example use is [here](https://github.com/Fantom-foundation/Sonic/blob/668e297b7a2aa24b8c538f0ebaf2c4cf4d1b1684/gossip/evmstore/store_tx.go#L55).